core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -350,9 +367,11 @@ public void commit() {
               TableMetadata updated = internalApply();
               ops.commit(base, updated);
             });
-    LOG.info("Committed snapshot changes");
+    LOG.info(
+        "Committed snapshot changes and prepare to clean up files at level={}",
+        cleanupLevel.name());
 
-    if (cleanExpiredFiles && !base.snapshots().isEmpty()) {
```

Comment:
Minor: I would've preferred for this to just change to cleanupLevel != CleanupLevel.NONE instead of nesting the logic further inside cleanExpiredSnapshots. Now someone has to read cleanExpiredSnapshots() to see it gets skipped when it's none, whereas I feel like it's better if cleanExpiredSnapshots just does cleanup, and we skip before that point.